appengine/firebase-tictactoe/src/main/java/com/example/appengine/firetactoe/Game.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.appengine.firetactoe;
+
+import com.google.common.io.CharStreams;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseOptions;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.googlecode.objectify.annotation.Entity;
+import com.googlecode.objectify.annotation.Id;
+
+import org.json.JSONObject;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.lang.RuntimeException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.regex.Pattern;
+
+@Entity
+public class Game {
+  static final Pattern[] XWins =
+      {Pattern.compile("XXX......"), Pattern.compile("...XXX..."), Pattern.compile("......XXX"),
+          Pattern.compile("X..X..X.."), Pattern.compile(".X..X..X."),
+          Pattern.compile("..X..X..X"), Pattern.compile("X...X...X"),
+          Pattern.compile("..X.X.X..")};
+  static final Pattern[] OWins =
+      {Pattern.compile("OOO......"), Pattern.compile("...OOO..."), Pattern.compile("......OOO"),
+          Pattern.compile("O..O..O.."), Pattern.compile(".O..O..O."),
+          Pattern.compile("..O..O..O"), Pattern.compile("O...O...O"),
+          Pattern.compile("..O.O.O..")};
+
+  @Id
+  public String id;
+  public String userX;
+  public String userO;
+  public String board;
+  public Boolean moveX;
+  public String winner;
+  public String winningBoard;
+
+  private static final String SERVICE_ACCOUNT_PATH = "resources/credentials.json";
+  private static final String FIREBASE_SNIPPET_PATH = "resources/firebase_config.html";
+  private static String sFirebaseDbUrl;
+  protected static String sFirebaseSnippet;
+
+  static {
+    try {
+      sFirebaseSnippet = CharStreams.toString(
+          new InputStreamReader(new FileInputStream(FIREBASE_SNIPPET_PATH)));
+      sFirebaseDbUrl = parseFirebaseUrl(sFirebaseSnippet);
+
+      FirebaseOptions options = new FirebaseOptions.Builder()
+          .setServiceAccount(new FileInputStream(SERVICE_ACCOUNT_PATH))
+          .setDatabaseUrl(sFirebaseDbUrl)
+          .build();
+      FirebaseApp.initializeApp(options);
+    } catch (FileNotFoundException e) {
```

Comment:
Honestly, after closer look at this, what you are currently doing, while not "good" is probably the simplest way to do it. Any time there could be an exception at object construction time is a tough one to deal with.

What I'd actually recommend is keep the singleton pattern you have, but then lazily initialize firebaseDbUrl in endFirebaseMessage. Maybe throw a kind of IOException instead of runtime exception so the servlet can show an error that Firebase isn't configured.